diff --git a/site.rb b/site.rb
index 250596bf..c0ea598f 100644
--- a/site.rb
+++ b/site.rb
@@ -224,10 +224,12 @@ module Jekyll
       regex = ''
       unless disabled
         @exclude.each do |x|
-          regex += "(?!#{x})"
+          escaped_x = Regex.escape(x)
+          regex += "(?!#{escaped_x})"
         end
         @languages.each do |x|
-          regex += "(?!#{x}\/)"
+          escaped_x = Regex.escape(x)
+          regex += "(?!#{escaped_x}\/)"
         end
       end
       start = disabled ? 'ferh' : 'href'
@@ -242,10 +244,12 @@ module Jekyll
       regex = ''
       unless disabled
         @exclude.each do |x|
-          regex += "(?!#{x})"
+          escaped_x = Regex.escape(x)
+          regex += "(?!#{escaped_x})"
         end
         @languages.each do |x|
-          regex += "(?!#{x}\/)"
+          escaped_x = Regex.escape(x)
+          regex += "(?!#{escaped_x}\/)"
         end
       end
       start = disabled ? 'ferh' : 'href'
